src/messages/MStatfs.h

```diff
@@ -22,13 +22,13 @@
 class MStatfs : public PaxosServiceMessage {
 
   static const int HEAD_VERSION = 2;
-  static const int COMPAT_VERSION = 0;
+  static const int COMPAT_VERSION = 1;
 
 public:
   uuid_d fsid;
   boost::optional<int64_t> data_pool;
 
-  MStatfs() : PaxosServiceMessage(CEPH_MSG_STATFS, 0, HEAD_VERSION) {}
+  MStatfs() : PaxosServiceMessage(CEPH_MSG_STATFS, HEAD_VERSION, COMPAT_VERSION) {}
```

Comment:
I'm not really clear on the difference between `enc_version` and `v` in:

`PaxosServiceMessage(int type, version_t v, int enc_version=1, int compat_enc_version=0)`

but based on other messages it looks like you want:

`PaxosServiceMessage(CEPH_MSG_STATFS, 0, HEAD_VERSION, COMPAT_VERSION)`

?